src/customer-service/Actions/ShowSingleIncidentAction.cs

```diff
@@ -0,0 +1,85 @@
+﻿using customer_service.Models;
+using System;
+using System.Globalization;
+
+namespace customer_service
+{
+    /**
+     * Class: ShowSingleIncidentAction
+     * Purpose: Displays the menu and details for the single incident report
+     * Author: Dayne Wright
+     * Methods:
+     *     void ReadInput() - Shows the incident report and prompts for a resolution if one is not provided. 
+     *     Then saves that resolution with the current date.
+     */
+
+    public class ShowSingleIncidentAction
+    {
+        /**
+         * Purpose: Shows incident report and saves resolution if needed.
+         * Arguments:
+         *     NONE
+         * Return:
+         *     Void
+         */
+        public static void ReadInput()
+        {
+            Customer customer = CustomerFactory.Instance.ActiveCustomer;
+            Incident incident = IncidentFactory.Instance.ActiveIncident;
+
+            if (customer == null || incident == null)
+            {
+                Console.WriteLine("A customer or incident could not be found.");
+                Console.Write("> ");
+                Console.ReadLine();
+            }
+            else
+            {
+                LabelFactory labelFactory = new LabelFactory();
+                IncidentTypeFactory typesFactory = new IncidentTypeFactory();
+
+                var labels = labelFactory.GetLabels(incident.IncidentTypeId);
+                var types = typesFactory.getAll();
+
+                Console.Write("\n");
+                Console.WriteLine("INCIDENT");
+                Console.WriteLine("==============================================================");
+                Console.WriteLine($"Customer: {customer.FirstName}, {customer.LastName}                         Order: {incident.OrderId}");
+                Console.WriteLine($"Incident Type: {types[incident.IncidentTypeId].Label}");
+                Console.Write("\n");
+                Console.WriteLine("Labels:");
+                foreach (Label label in labels)
+                {
+                    Console.WriteLine($"* {label.Description}");  
+                }
+                Console.Write("\n");
+
+                if (incident.Resolution == "" || incident.Resolution == null)
+                {
+                    Console.WriteLine("==============================================================");
+                    Console.WriteLine("Enter Resolution: ");
+                    Console.Write("> ");
+
+                    incident.Resolution = Console.ReadLine();
```

Comment:
I'm not sure if this can be avoided 100%. You _could_ also enter a resolution like so and it'd run just fine:

```sql
Delivered product to customer' DROP TABLE Employee
```

We **could** parse the string for `'` or `"` characters and append a `\` before them to escape the character. Or a `"` to escape characters if the SQL query is in a string literal.